### ISSUE TITLE
No sign of kernel initialisation on Spike

### ISSUE THREAD

**Issue by OrkunAliOzkan** [riscv/riscv-isa-sim#1359]
I am attempting to boot OpenSBI+linux on the Spike simulator, and have been making good use of documentation https://github.com/riscv-software-src/opensbi/blob/master/docs/platform/spike.md

I built my kernel image:
```
make ARCH=riscv CROSS_COMPILE=riscv64-linux-gnu- defconfig
make ARCH=riscv CROSS_COMPILE=riscv64-linux-gnu-
```

And embedded it into the OpenSBI executable:
```
make PLATFORM=generic CROSS_COMPILE=riscv64-linux-gnu- FW_PAYLOAD_PATH=/path/to/kernel/Image
```

Then proceeded to run the simulator:
```
spike -m256 \
    --isa=RV64IMAFDC \
    --bootargs 'root=/dev/ram rw console=hvc0 earlycon=sbi' \
    path/to/the/fw_payload.elf \
    --kernel=/path/to/kernel/Image
```

Which results in console log output:
```
OpenSBI v1.2-105-g2868f26
   ____                    _____ ____ _____
  / __ \                  / ____|  _ \_   _|
 | |  | |_ __   ___ _ __ | (___ | |_) || |
 | |  | | '_ \ / _ \ '_ \ \___ \|  _ < | |
 | |__| | |_) |  __/ | | |____) | |_) || |_
  \____/| .__/ \___|_| |_|_____/|___/_____|
        | |
        |_|

Platform Name             : ucbbar,spike-bare
Platform Features         : medeleg
Platform HART Count       : 1
Platform IPI Device       : aclint-mswi
Platform Timer Device     : aclint-mtimer @ 10000000Hz
Platform Console Device   : uart8250
Platform HSM Device       : ---
Platform PMU Device       : ---
Platform Reboot Device    : htif
Platform Shutdown Device  : htif
Platform Suspend Device   : ---
Platform CPPC Device      : ---
Firmware Base             : 0x80000000
Firmware Size             : 232 KB
Firmware RW Offset        : 0x20000
Runtime SBI Version       : 1.0

Domain0 Name              : root
Domain0 Boot HART         : 0
Domain0 HARTs             : 0*
Domain0 Region00          : 0x0000000080000000-0x000000008001ffff M: (R,X) S/U: ()
Domain0 Region01          : 0x0000000000020000-0x000000008003ffff M: (R,X) S/U: ()
Domain0 Region02          : 0x0000000002080000-0x00000000020bffff M: (I,R,W) S/U: ()
Domain0 Region03          : 0x0000000002000000-0x000000000207ffff M: (I,R,W) S/U: ()
Domain0 Region04          : 0x0000000000000000-0xffffffffffffffff M: (R,W,X) S/U: (R,W,X)
Domain0 Next Address      : 0x0000000080200000
Domain0 Next Arg1         : 0x0000000082200000
Domain0 Next Mode         : S-mode
Domain0 SysReset          : yes
Domain0 SysSuspend        : yes

Boot HART ID              : 0
Boot HART Domain          : root
Boot HART Priv Version    : v1.12
Boot HART Base ISA        : rv64imafdc
Boot HART ISA Extensions  : none
Boot HART PMP Count       : 16
Boot HART PMP Granularity : 4
Boot HART PMP Address Bits: 54
Boot HART MHPM Count      : 0
Boot HART MIDELEG         : 0x0000000000000222
Boot HART MEDELEG         : 0x00000000000b109

```

Which clearly shows the kernel initalising is not presented, if running at all. I know the contents of the firmware can run when built with this cross compiler, because I attempted to run opensbi on the sim with the test payload only and it ran, however I am unsure of how to diagnose this problem and proceed further. Any insight would be greatly appreciated.

For those interested in replicating my steps:
Busybox: v1.36.0
Linux: https://github.com/litex-hub/linux/tree/litex-rebase
OpenSBI: 2868f26131308ff345382084681ea89c5b0159f1
Toolchain: riscv64-unknown-elf-gcc-12.1.0

**Comment by OrkunAliOzkan** [comment]
I believe the primary issue is the fact that I am not passing in spikes dtb, would anyone know where to find it?

EDIT:
I generated the dtb from the dts produced by invoking the command
```
spike --dump-dts foo > spike_dts.dts
```

Produces following dts:
```
/dts-v1/;

/ {
  #address-cells = <2>;
  #size-cells = <2>;
  compatible = "ucbbar,spike-bare-dev";
  model = "ucbbar,spike-bare";
  chosen {
    stdout-path = &SERIAL0;
    bootargs = "console=ttyS0 earlycon";
  };
  cpus {
    #address-cells = <1>;
    #size-cells = <0>;
    timebase-frequency = <10000000>;
    CPU0: cpu@0 {
      device_type = "cpu";
      reg = <0>;
      status = "okay";
      compatible = "riscv";
      riscv,isa = "rv64imafdc_zicntr_zihpm";
      mmu-type = "riscv,sv57";
      riscv,pmpregions = <16>;
      riscv,pmpgranularity = <4>;
      clock-frequency = <1000000000>;
      CPU0_intc: interrupt-controller {
        #address-cells = <2>;
        #interrupt-cells = <1>;
        interrupt-controller;
        compatible = "riscv,cpu-intc";
      };
    };
  };
  memory@80000000 {
    device_type = "memory";
    reg = <0x0 0x80000000 0x0 0x80000000>;
  };
  soc {
    #address-cells = <2>;
    #size-cells = <2>;
    compatible = "ucbbar,spike-bare-soc", "simple-bus";
    ranges;
    clint@2000000 {
      compatible = "riscv,clint0";
      interrupts-extended = <&CPU0_intc 3 &CPU0_intc 7 >;
      reg = <0x0 0x2000000 0x0 0xc0000>;
    };
    PLIC: plic@c000000 {
      compatible = "riscv,plic0";
      #address-cells = <2>;
      interrupts-extended = <&CPU0_intc 11 &CPU0_intc 9 >;
      reg = <0x0 0xc000000 0x0 0x1000000>;
      riscv,ndev = <0x1f>;
      riscv,max-priority = <0xf>;
      #interrupt-cells = <1>;
      interrupt-controller;
    };
    SERIAL0: ns16550@10000000 {
      compatible = "ns16550a";
      clock-frequency = <10000000>;
      interrupt-parent = <&PLIC>;
      interrupts = <1>;
      reg = <0x0 0x10000000 0x0 0x100>;
      reg-shift = <0x0>;
      reg-io-width = <0x1>;
    };
  };
  htif {
    compatible = "ucb,htif0";
  };
};
```

Obtaining kernel panic:
```
OpenSBI v1.2-105-g2868f26
   ____                    _____ ____ _____
  / __ \                  / ____|  _ \_   _|
 | |  | |_ __   ___ _ __ | (___ | |_) || |
 | |  | | '_ \ / _ \ '_ \ \___ \|  _ < | |
 | |__| | |_) |  __/ | | |____) | |_) || |_
  \____/| .__/ \___|_| |_|_____/|___/_____|
        | |
        |_|

Platform Name             : ucbbar,spike-bare
Platform Features         : medeleg
Platform HART Count       : 1
Platform IPI Device       : aclint-mswi
Platform Timer Device     : aclint-mtimer @ 10000000Hz
Platform Console Device   : uart8250
Platform HSM Device       : ---
Platform PMU Device       : ---
Platform Reboot Device    : htif
Platform Shutdown Device  : htif
Platform Suspend Device   : ---
Platform CPPC Device      : ---
Firmware Base             : 0x80000000
Firmware Size             : 232 KB
Firmware RW Offset        : 0x20000
Runtime SBI Version       : 1.0

Domain0 Name              : root
Domain0 Boot HART         : 0
Domain0 HARTs             : 0*
Domain0 Region00          : 0x0000000080000000-0x000000008001ffff M: (R,X) S/U: ()
Domain0 Region01          : 0x0000000080020000-0x000000008003ffff M: (R,W) S/U: ()
Domain0 Region02          : 0x0000000002080000-0x00000000020bffff M: (I,R,W) S/U: ()
Domain0 Region03          : 0x0000000002000000-0x000000000207ffff M: (I,R,W) S/U: ()
Domain0 Region04          : 0x0000000000000000-0xffffffffffffffff M: (R,W,X) S/U: (R,W,X)
Domain0 Next Address      : 0x0000000080200000
Domain0 Next Arg1         : 0x0000000082200000
Domain0 Next Mode         : S-mode
Domain0 SysReset          : yes
Domain0 SysSuspend        : yes

Boot HART ID              : 0
Boot HART Domain          : root
Boot HART Priv Version    : v1.12
Boot HART Base ISA        : rv64imafdc
Boot HART ISA Extensions  : none
Boot HART PMP Count       : 16
Boot HART PMP Granularity : 4
Boot HART PMP Address Bits: 54
Boot HART MHPM Count      : 0
Boot HART MIDELEG         : 0x0000000000000222
Boot HART MEDELEG         : 0x000000000000b109
[    0.000000] Linux version 6.3.0-g68cfb31f2488-dirty (root@5ed389c0e6e0) (riscv64-linux-gnu-gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #2 SMP Fri May 12 09:52:43 UTC 2023
[    0.000000] Machine model: ucbbar,spike-bare
[    0.000000] earlycon: ns16550a0 at MMIO 0x0000000010000000 (options '')
[    0.000000] printk: bootconsole [ns16550a0] enabled
[    0.000000] efi: UEFI not found.
[    0.000000] OF: reserved mem: 0x0000000080000000..0x000000008001ffff (128 KiB) map non-reusable mmode_resv0@80000000
[    0.000000] OF: reserved mem: 0x0000000080020000..0x000000008003ffff (128 KiB) map non-reusable mmode_resv1@80020000
[    0.000000] Oops - store (or AMO) access fault [#1]
[    0.000000] Modules linked in:
[    0.000000] CPU: 0 PID: 0 Comm: swapper Not tainted 6.3.0-g68cfb31f2488-dirty #2
[    0.000000] Hardware name: ucbbar,spike-bare (DT)
[    0.000000] epc : __memset+0x60/0xfc
[    0.000000]  ra : create_pgd_mapping+0x108/0x19c
[    0.000000] epc : ffffffff8080e0f0 ra : ffffffff80a06b4e sp : ffffffff81403dd0
[    0.000000]  gp : ffffffff814fc150 tp : ffffffff8140dac0 t0 : ff1bfffffebfc000
[    0.000000]  t1 : 0000000082200a96 t2 : 7265736572203a46 s0 : ffffffff81403e20
[    0.000000]  s1 : ff60000000000000 a0 : ff1bfffffebfc000 a1 : 0000000000000000
[    0.000000]  a2 : 0000000000001000 a3 : ff1bfffffebfd000 a4 : 0000000000000000
[    0.000000]  a5 : ff1bfffffebfc000 a6 : 00000000000005c6 a7 : 0000000000000048
[    0.000000]  s2 : 0000000080000000 s3 : 0000000000200000 s4 : 00000000000000e7
[    0.000000]  s5 : ffffffff814fb978 s6 : 00000000fffff000 s7 : ffffffff81508b00
[    0.000000]  s8 : ffffffff80c01000 s9 : 0000ffffffffffff s10: 00000000001fffff
[    0.000000]  s11: 0000000000000001 t3 : 0000000100000000 t4 : ffffffffffffffff
[    0.000000]  t5 : ffffffff80c0c938 t6 : ffffffff80c0c950
[    0.000000] status: 0000000200000100 badaddr: ff1bfffffebfc000 cause: 0000000000000007
[    0.000000] [<ffffffff8080e0f0>] __memset+0x60/0xfc
[    0.000000] [<ffffffff80a06f62>] create_linear_mapping_range+0xda/0x12e
[    0.000000] [<ffffffff80a07e3e>] paging_init+0x42a/0x5b8
[    0.000000] [<ffffffff80a03458>] setup_arch+0xd0/0x594
[    0.000000] [<ffffffff80a00742>] start_kernel+0x8a/0x746
[    0.000000] Code: 1007 82b3 40e2 0797 0000 8793 00e7 8305 97ba 8782 (b023) 00b2 
[    0.000000] ---[ end trace 0000000000000000 ]---
[    0.000000] Kernel panic - not syncing: Fatal exception in interrupt
(spike) q
```

Relevant reading material: 
Store access fault: https://forums.sifive.com/t/what-is-the-difference-between-access-fault-and-page-fault/1473
https://groups.google.com/a/groups.riscv.org/g/sw-dev/c/f1I97HMBulc

**Comment by OrkunAliOzkan** [comment]
I was being dumb, if I hadn't only allocated 256MiB of target memory the entire system would have worked....
TLDR: remove -m256
